Core/InputOutput/TiffHandler.cpp

```
@@ -107,27 +119,10 @@ void TiffHandler::read_data()
 {
     assert(m_tiff);
 
-    uint16 sampleFormat(0);
-    if (!TIFFGetField(m_tiff, TIFFTAG_SAMPLEFORMAT, &sampleFormat))
-        sampleFormat = 1;
-
-    static_assert(
-          supported_sampleSize == sizeof(uint32) &&
-          supported_sampleSize == sizeof(int32)  &&
-          supported_sampleSize == sizeof(float),
-          "sanity check failed");
-
-    switch (sampleFormat) {
-    case 1: // unsigned int
-    case 2: // signed int
-    case 3: // IEEE float
-      break;
-    default:
-      throw Exceptions::FormatErrorException("TiffHandler: unexpected sample format");
-    }
-
+    assert(0 == m_bitsPerSample%8);
```

This assert is a bit confusing here. First I thought it is the only check existing, and it is better to replace it with exception throw (to keep GUI alive in the case of unexpected format). Then I saw another check for bitPerSample above, with proper exception throw. So no harm from assert here.